Bluetooth: host: Refactor address and buf to new source files

joerchan opened the pull request:
Refactor bluetooth  address and buf handling out from hci_core and to new source files matching the headers these functions are defined in.

Document the addr.h functions and defines.

joerchan committed:
Bluetooth: host: Refactor out address handling to new file

Refactor implementation of functions defined in addr.h to its own
source file addr.c.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
Bluetooth: host: Refactor out buffer handling to new file

Refactor out implementation of functions defined in buf.h to its own
source file buf.c

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
Bluetooth: host: Document functions and macros in addr.h

Document helper functions and definitions in addr.h header.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

jhedberg commented:
@joerchan there's a compliance issue that needs fixing (redundant blank line at the end of a file)

joerchan committed:
Bluetooth: host: Make acl_data extension of bt_buf user data explicit

Clarify that the acl_data struct includes the bt_buf type as the
first variable in the user data struct.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan commented:
> @joerchan there's a compliance issue that needs fixing (redundant blank line at the end of a file)

@jhedberg Fixed that.

There was however another issue with ACL_FLOW_CONTROL. I have now moved the declaration of acl_in_pool to buf.c as well.
But that meant I had to expose the host num complete command function and the definition of the acl_data struct.

Also I was a bit puzzled as to how the acl_data type variable worked, so when I figured it out I made another change to it that I think is a bit clearer.

Thalley commented on include/bluetooth/addr.h:
```diff
@@ -40,44 +41,89 @@ typedef struct {
 	bt_addr_t a;
 } bt_addr_le_t;
 
+/* Bluetooth device "any" address, not a valid address */
```
Do we want these descriptions to be caught by Doxygen? Shoulnd't they start with `/**` then (Similar to the additional documentation you added in `uuid.h`).

Thalley commented on include/bluetooth/addr.h:
```diff
@@ -40,44 +41,89 @@ typedef struct {
+ *  @param a First Bluetooth device address to compare
+ *  @param b Second Bluetooth device address to compare
+ *
+ *  @return negative value if @a a < @a b, 0 if @a a == @a b, else positive
```
Hah, this line is so hard to follow with the `@a` references :D (Not wrong, at all, but just fun with all the `@a` interleaved)

Thalley commented on include/bluetooth/addr.h:
```diff
@@ -40,44 +41,89 @@ typedef struct {
+/** @brief Create a Bluetooth LE random static address. */
 int bt_addr_le_create_static(bt_addr_le_t *addr);
 
+/** @brief Check if a Bluetooth LE address is a random private resolvable
+ *         address.
```
If you wanted to get this on a single line, you could use the RPA abbreviation here, and then expand it on the next line

Thalley commented on include/bluetooth/addr.h:
```diff
@@ -87,6 +133,15 @@ static inline bool bt_addr_le_is_rpa(const bt_addr_le_t *addr)
 	return BT_ADDR_IS_RPA(&addr->a);
 }
 
+/** @brief Check if a Bluetooth LE address is valid identity address.
+ *
```
There's some incoherence with
```
/** @brief 
```
and
```
/**
 * @brief
```

in this PR. 

I'm no longer sure what the ideal way is, since `bluetooth.h` was refactored in that regards lately, but some of the other Bluetooth includes weren't...

Thalley reviewed (approved):
A few documentation comments that does not need to be addressed, but probably should be considered.

jhedberg commented:
@joerchan there are some build tests that seem to be failing..
